### ISSUE TITLE
Documentation for standalone function compilation

### ISSUE THREAD

**Issue by martincerny** [stan-dev/stan#2329]
#### Summary:
Write documentation for standalone function compilation.

#### Description:
When #2267 was implemented and merged, only rudimentary documentation was created. Some description in both manual and the stanc command line is desirable.

#### Current Version:
v2.15.0

**Comment by bob-carpenter** [comment]
We don't talk about compilation or the interfaces at all in the Stan manual.    But we can talk about the fact that it's possible to compile out just functions as well as models.

This should probably be an issue in each of the interfaces.  @seantalts --- any way we can broadcast issues to all the interfaces?

**Comment by seantalts** [comment]
I don't know of a way - there are some issue movers but not broadcasters. :(

**Comment by martincerny** [comment]
Yeah, it probably is better suited for the CmdStan manual than for the main manual. For other interfaces, it is necessary to make this feature accesible first. I am planning to integrate with RStan (https://github.com/stan-dev/rstan/issues/422), but I have limited experience with Python, Julia etc. so that will be left for others.
